.github/workflows/build-docker.yml

```
@@ -0,0 +1,143 @@
+name: Build and Push Docker Image
+
+on:
+  push:
+    branches:
+      - main
+    tags:
+      - 'v*.*.*'
```

```suggestion
      - '*.*.*'
```
We typically don't tag starting with `v`.

---

:thinking: we might maybe should but I will change it.